### PR TITLE
Fix all PythonScriptsTest for python3

### PR DESCRIPTION
**Pull request by rosswhitfield**
This **does not** fix all the files in `scripts` for python 3, just the ones that are needed to get the `PythonScripts` unit tests to pass.

**To test:**
- Code review
- Compile with python3, then run `ctest -R PythonScriptsTest`, they should all pass

_Does not need to be in the release notes._

cc: @peterfpeterson @martyngigg 

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/SANS/isis_reduction_steps.py [hunk 1]
```diff
@@ -1,11 +1,12 @@
-﻿# pylint: disable=too-many-lines, too-many-branches, invalid-name, super-on-old-class, protected-access,
+# pylint: disable=too-many-lines, too-many-branches, invalid-name, super-on-old-class, protected-access,
 # pylint: disable=too-few-public-methods,too-few-public-methods, too-many-arguments, too-many-instance-attributes
 """
     This file defines what happens in each step in the data reduction, it's
     the guts of the reduction. See ISISReducer for order the steps are run
     in and the names they are given to identify them
 
-    Most of this code is a copy-paste from SANSReduction.py, organized to be used with
+    Most of this code is a copy-paste from __future__ import (absolute_import, division, print_function)
```
Comment (martyngigg) [inline scripts/SANS/isis_reduction_steps.py]: It looks like the `__future__` imports have not gone to the right place in the file.

File: scripts/CryPowderISIS/cry_focus.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-﻿# pylint: disable=too-many-arguments,unused-variable
+# pylint: disable=too-many-arguments,unused-variable
```
Comment (martyngigg) [inline scripts/CryPowderISIS/cry_focus.py]: Should there be a `__future__` import here?

File: scripts/Interface/ui/reflectometer/refl_gui.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-﻿# pylint: disable = too-many-lines, invalid-name, line-too-long, too-many-instance-attributes, too-many-branches,too-many-locals, too-many-nested-blocks
+# pylint: disable = too-many-lines, invalid-name, line-too-long, too-many-instance-attributes, too-many-branches,too-many-locals, too-many-nested-blocks
```
Comment (martyngigg) [inline scripts/Interface/ui/reflectometer/refl_gui.py]: Should there be a `__future__` import here

File: scripts/QECoverage.py [hunk 4]
```diff
@@ -1,4 +1,4 @@
-﻿# pylint: disable=line-too-long, too-many-instance-attributes, invalid-name, missing-docstring, too-many-statements
+# pylint: disable=line-too-long, too-many-instance-attributes, invalid-name, missing-docstring, too-many-statements
 # pylint: disable= too-many-branches, no-self-use
 import sys
```
Comment (martyngigg) [inline scripts/QECoverage.py]: Should there be a __future__ import here

**Review by martyngigg — CHANGES_REQUESTED** [review]

File: scripts/Inelastic/Direct/RunDescriptor.py [hunk 5]
```diff
@@ -1339,7 +1340,7 @@ def _split_ws_name(self,ws_name):
 # Hell knows how to redefine these warnings or if they are valid or not
 #pylint: disable=W0141
 #pylint: disable=W0110
-            self._ws_cname = part_ind + filter(lambda c: not c.isdigit(), name)
+            self._ws_cname = part_ind + ''.join(re.findall('\d+', name))
```
Comment (rosswhitfield) [inline scripts/Inelastic/Direct/RunDescriptor.py]: Whoops I missed the `not` and assumed it was the same the other `filter` cases I changed.

**Comment by rosswhitfield** [comment]
In regards to adding `__future__` imports to the other files, I only wanted to add them to the files I spend time converting to python 3, _i.e._ the ones required for the unit tests to pass, so that it easy to tell which ones still needed working on. After this pull request only 65 out of 410 of the `*.py` files in `scripts` will have the `__future__` imports, there is still a lot of work remaining.

**Review by martyngigg — APPROVED** [review]

**Comment by martyngigg** [comment]
This looks good to go now.

@mantidproject/gatekeepers let's :shipit: before conflicts start arising.

**Pull request merged**